### PR TITLE
Add some more docs.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: src/Test/StateMachine/Internal/AlphaEquality.hs [hunk 1]
```diff
@@ -87,24 +92,28 @@ canonicalFork (Fork l p r) = Fork l' p' r'
   l'        = fst $ canonical' im' l
   r'        = fst $ canonical' im' r
 
+-- | Check if two lists of commands are equal modulo
+--   \(\alpha\)-convertion.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/AlphaEquality.hs]: conversion

File: src/Test/StateMachine/Internal/AlphaEquality.hs [hunk 2]
```diff
@@ -87,24 +92,28 @@ canonicalFork (Fork l p r) = Fork l' p' r'
...
   => Eq (IntRefed cmd)
-  => [IntRefed cmd]
-  -> [IntRefed cmd]
+  => [IntRefed cmd]     -- ^ The two
+  -> [IntRefed cmd]     -- ^ input lists.
   -> Bool
 alphaEq c0 c1 = canonical c0 == canonical c1
 
+-- | Check if two forks of commands are equal modulo
+--   \(\alpha\)-convertion.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/AlphaEquality.hs]: conversion

File: src/Test/StateMachine/Internal/Parallel.hs [hunk 3]
```diff
@@ -261,6 +268,8 @@ runMany kit sem = flip foldM () $ \_ cmd'@(IntRefed cmd iref) -> do
     atomically $ writeTChan (getHistoryChannel kit) $
       ResponseEvent (toDyn resp) (getProcessIdHistory kit)
 
+-- | Lift the semantics of a single typed command into a semantics for
+--   forks of untyped internal commands.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Parallel.hs]: Mention `History`?

File: src/Test/StateMachine/Internal/Sequential.hs [hunk 4]
```diff
@@ -66,28 +69,37 @@ import           Test.StateMachine.Types
 
 ------------------------------------------------------------------------
 
+-- | Lift a generator of untyped commands with reference placeholders
+--   into a generator of lists of untyped internal commands.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: I'm not sure if we want to really call them `untyped`, but maybe that is okay.

File: src/Test/StateMachine/Internal/Sequential.hs [hunk 5]
```diff
@@ -148,12 +164,14 @@ liftShrink shrinker = go
...
+-- | Remove commands that uses a reference.
 removeCommands
   :: forall cmd
   .  IxFoldable cmd
   => HasResponse cmd
-  => IntRefed cmd
-  -> [IntRefed cmd]
+  => IntRefed cmd    -- ^ If this command returns a reference, then
+  -> [IntRefed cmd]  -- ^ remove all commands that use that reference in this
+                     -- list.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Sequential.hs]: recursively

File: src/Test/StateMachine/Internal/Types.hs [hunk 6]
```diff
@@ -18,6 +18,8 @@
 -- Stability   :  provisional
 -- Portability :  non-portable (GHC extensions)
 --
+-- This module defins
+--
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Types.hs]: We should probably not encourage [defin](https://en.wikipedia.org/wiki/Shark_fin_soup#Demand_declines.2C_2005-present) of any kind.

File: src/Test/StateMachine/Internal/Types.hs [hunk 7]
```diff
@@ -49,16 +51,20 @@ import           Test.StateMachine.Types
 
 ------------------------------------------------------------------------
 
+-- | Type-level function that maybe returns a reference.
 type family MayResponse_ (refs :: TyFun ix k -> Type) (resp :: Response ix) :: k where
   MayResponse_ refs ('Response  t) = ()
   MayResponse_ refs ('Reference i) = refs @@ i
 
+-- | Internal untyped commands.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Types.hs]: Some say untyped, some say super typed... ;P

File: src/Test/StateMachine/Internal/Types.hs [hunk 8]
```diff
@@ -49,16 +51,20 @@ import           Test.StateMachine.Types
...
 data IntRefed (f :: Signature ix) where
   IntRefed :: ( Show     (Response_ ConstIntRef resp)
               , Typeable (Response_ ConstIntRef resp)
               , Typeable resp
               ) => f ConstIntRef resp -> MayResponse_ ConstIntRef resp -> IntRefed f
 
+-- | Forks are used to represent parallel programs. They have a sequential
+-- prefix, the middle argument, and two parallel branches.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Types.hs]: This reads as if they have three things:
* sequential prefix
* middle argument
* two parallel branches

File: src/Test/StateMachine/Internal/Types.hs [hunk 9]
```diff
@@ -73,10 +79,12 @@ instance Pretty a => Pretty (Fork a) where
 
 ------------------------------------------------------------------------
 
-showFork:: (a -> String) -> Fork a -> String
+-- | Explicit show function for forks.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Types.hs]: Explicit? I kind of understand what you mean but..

File: src/Test/StateMachine/Internal/Types/IntRef.hs [hunk 10]
```diff
@@ -26,13 +28,20 @@ import           Data.Singletons.Prelude
...
 ------------------------------------------------------------------------
 
+-- | An internal (or integer) reference consists of a reference and a
+--   process id.
 data IntRef = IntRef Ref Pid
   deriving (Eq, Ord, Show, Read)
 
+-- | A process id is merely a natural number that keeps track of which
+--   thread the reference comes from. In the sequential case the process
+--   id is 0.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Types/IntRef.hs]: And in the case of prefix for parallel.

File: src/Test/StateMachine/Internal/Utils.hs [hunk 11]
```diff
@@ -44,18 +47,23 @@ import           Test.QuickCheck.Property
 
 ------------------------------------------------------------------------
 
+-- | The type of a shrinker function.
 type Shrinker a = a -> [a]
 
+-- | Keep generating until we actually get a value.
```
Comment (Danten) [inline src/Test/StateMachine/Internal/Utils.hs]: Wait when did this function change?

File: src/Test/StateMachine/Internal/Sequential.hs (hunk 4 above)
Comment (stevana) [inline src/Test/StateMachine/Internal/Sequential.hs]: Kept calling them `untyped`, but I'm happy to change it. My reasoning was that `resp` is hidden...

File: src/Test/StateMachine/Internal/Utils.hs (hunk 11 above)
Comment (stevana) [inline src/Test/StateMachine/Internal/Utils.hs]: Do you think it's worth opening up a ticket for this?

**Comment by stevana** [comment]
Fixed most stuff, thanks!

**Pull request merged**